src/structures/Guild.js

```diff
@@ -392,19 +392,23 @@ class Guild extends Base {
    */
 
   /**
-   * Fetches a collection of banned users in this guild.
-   * @returns {Promise<Collection<Snowflake, BanInfo>>}
-   */
-  fetchBans() {
-    return this.client.api.guilds(this.id).bans.get().then(bans =>
-      bans.reduce((collection, ban) => {
+   * Fetches a single banned user or a collection of banned users in this guild.
+   * @param {string} [id] Optional id of a banned user.
+   * @returns {Promise<Collection<Snowflake, BanInfo>> | Promise<BanInfo>}
+   */
+  fetchBans(id) {
+    return this.client.api.guilds(this.id).bans.get(id).then(data => {
+      if (id) {
+        return { reason: data.reason, user: this.client.users.add(data.user) };
+      }
+      return data.reduce((collection, ban) => {
         collection.set(ban.user.id, {
-          reason: ban.reason,
-          user: this.client.users.add(ban.user),
+          reason: data.reason,
```

Comment:
This should stay `ban` and not `data`, which is the raw array received from the api.